src/query/service/src/pipelines/processors/transforms/transform_udf_server.rs

```diff
@@ -237,9 +257,9 @@ impl AsyncTransform for TransformUdfServer {
 
             let task_len = tasks.len() as u64;
             record_running_requests_external_start(func.name.clone(), task_len);
-            let blocks = futures::future::join_all(tasks).await;
+            let task_results = futures::future::join_all(tasks).await;
```

Comment:
maybe use `futures::stream::iter(tasks).for_each_concurrent`

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b737a80a798e0f603be01fbb7caeb09f

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4435eb4674a69b2b68f5c59cde7bc5ca

Reply:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=04620f7a98d4ebb9496e9086cced031f

I prefer this one.

Reply:
But we may need a global permit during udf processors * thread, so the second one can meet our needs.